cache.go

```diff
@@ -84,6 +88,25 @@ func (c *Cache) Dir(ppath string) (string, error) {
 	return dir, nil
 }
 
+// Dirs does the same as patsy.Dirs but cached.
+func (c *Cache) Dirs(ppath string) (map[string]string, error) {
+	// check the cache first
+	if dirs, ok := c.getDirs(ppath); ok {
+		return dirs, nil
+	}
+	dirs, err := Dirs(c.env, ppath)
+	if err != nil {
+		return nil, err
+	}
+	c.setDirs(ppath, dirs)
+
+	for importPath, dir := range dirs {
+		c.setDir(importPath, dir)
+		c.setPath(dir, importPath)
+	}
```

Comment:
I think it makes sense to fill the cache with the single paths found incase `Dirs` and `Dir` are both used?